### DISCUSSION TITLE
Problems with MQTT json

### DISCUSSION THREAD

**Discussion by simen13** [frangoteam/FUXA#991] [General]
Hi

I have set up a PLC to publish json-string MQTT messages to an MQTT-broker.
First I set up the json string to look like this:

{
  "Room": "LM_Frukting",
  "Type": "CO2-sensor",
  "sensor_id": "RY001_PV",
  "CO2": 1109
}

When I tried to read this in Fuxa, I could read all the variables.

It turned out though, that if I want to get the data logged into influxdb, I have to set up the json string like this (think it's an array..):

[
  {
    "CO2": 1109
  },
  {
    "Room": "LM_Frukting",
    "Type": "CO2-sensor",
    "sensor_id": "RY001_PV"
  }
]

Then it turns out that I can not read the variables anymore. Is there a way to solve this?

![image](https://github.com/frangoteam/FUXA/assets/148850779/ba9febf0-1d40-4c23-9f5c-6250e1c41de7)
![image](https://github.com/frangoteam/FUXA/assets/148850779/25531ff1-9fb5-4750-baa4-e53c1d3550de)

Thanks

**Comment by simen13** [comment]
Actually, I solved it by transforming the data in node-red instead of the PLC.

**Reply by robbudge** [reply]
What PLC are you utilizing. 
im testing with Codesys and have had good success  with sending individual tags in raw format to the broker and Fuxa. 

JSON format not so much, due you formatting issues at either end. 

Raw format also means less data is published. 
as the data change / publish can be determined in the PLC and not periodically all data.

**Comment by unocelli** [comment]
I would solve it with a script, if it is an array you will have an id to identify the topic and creating an UI to do this kind of configuration can be a challenge. you make a subscription of the array as object and in a script you bind the array value to server tags

**Comment by robbudge** [comment]
@unocelli , could you shine some light on  'script you bind the array value to server tags'
i am really in need of setting up a Server tag bind via script.

**Reply by unocelli** [reply]
Hi,
In this script I convert the raw value (read as string format) of mqtt topic (that contain the array) in a dictionary id/value and bind the a and c to FUXA server Tags
 
![mqtt-array-to-script](https://github.com/frangoteam/FUXA/assets/48684550/000c9f38-1b19-4d0a-ad94-88ea56a293b1)
